File: jambeth/jambeth-merge/src/test/java/de/osthus/ambeth/cache/mock/CacheFactoryMock.java
package de.osthus.ambeth.cache.mock;

import de.osthus.ambeth.cache.CacheFactoryDirective;
import de.osthus.ambeth.cache.ICacheFactory;
import de.osthus.ambeth.cache.IDisposableCache;

/**
 * Support for unit tests that do not include jAmbeth.Cache
 */
public class CacheFactoryMock implements ICacheFactory
{
	@Override
	public IDisposableCache createPrivileged(CacheFactoryDirective cacheFactoryDirective)
	{
		return null;
	}

	@Override
	public IDisposableCache createPrivileged(CacheFactoryDirective cacheFactoryDirective, boolean foreignThreadAware, Boolean useWeakEntries)
	{
		return null;
	}

	@Override
	public IDisposableCache create(CacheFactoryDirective cacheFactoryDirective)
	{
		return null;
	}

	@Override
	public IDisposableCache create(CacheFactoryDirective cacheFactoryDirective, boolean foreignThreadAware, Boolean useWeakEntries)
	{
		return null;
	}
}

File: ambeth/Ambeth.Merge.Test/ambeth/cache/mock/CacheFactoryMock.cs
using System;
namespace De.Osthus.Ambeth.Cache.Mock
{
    /**
     * Support for unit tests that do not include jAmbeth.Cache
     */
    public class CacheFactoryMock : ICacheFactory
    {
        public IDisposableCache Create(CacheFactoryDirective cacheFactoryDirective, String name)
        {
            return null;
        }

        public IDisposableCache Create(CacheFactoryDirective cacheFactoryDirective, bool foreignThreadAware, bool? useWeakEntries, String name)
        {
            return null;
        }

        public IDisposableCache CreatePrivileged(CacheFactoryDirective cacheFactoryDirective, String name)
        {
            return null;
        }

        public IDisposableCache CreatePrivileged(CacheFactoryDirective cacheFactoryDirective, bool foreignThreadAware, bool? useWeakEntries, String name)
        {
            return null;
        }
    }
}